felix/bpf-gpl/tc.c

```diff
@@ -606,16 +606,20 @@ static CALI_BPF_INLINE void calico_tc_process_ct_lookup(struct cali_tc_ctx *ctx)
 			goto skip_policy;
 		}
 		ctx->state->flags |= CALI_ST_DEST_IS_HOST;
-	} else if (CALI_F_FROM_HEP && !ctx->nat_dest && !cali_rt_is_local(dest_rt)) {
-		/* Disable FIB, let the packet go through the host after it is
-		 * policed. It is ingress into the system and we got a packet, which is
-		 * not for this host, and it wasn't resolved as a service and it is not
-		 * for a local workload either. But we hit a route so it may be some L2
-		 * broadcast, we do not quite know. Let the host route it or dump it.
-		 *
-		 * https://github.com/projectcalico/calico/issues/8918
-		 */
-		ctx->state->flags |= CALI_ST_SKIP_FIB;
+	} else if (CALI_F_FROM_HEP) {
+		if (cali_rt_flags_local_workload_vm(dest_rt->flags)) {
```

Comment:
This will be valid from WEP after https://github.com/projectcalico/calico/pull/9965 merges